### DISCUSSION TITLE
Hsid sub cookie for google.com needs to be deleted

### DISCUSSION THREAD

**Discussion by snowbound** [Cookie-AutoDelete/Cookie-AutoDelete#1080] [Q&A / Support] [answered]
In the past Gmail allowed one to uncheck a remember me box so that cookies were current session only. Google changed that a couple of years ago. I have found that there is a hsid subcookie for google,com that if one sets it to a session only cookie or deletes that cookie on browser exit/restart then the user is forcibly logged out of the account. It has to be only this cookie that is deleted because with Gmail there are also 2FA cookies that are stored elsewhere so that doing a mass deletion causes 2FA to have to be reentered every time.

I have tried the following

![image](https://user-images.githubusercontent.com/596147/111164883-73c22e80-856c-11eb-8d87-5852403155e6.png)

but nothing happens. I am still logged in. What am I missing?

**Comment by kennethtran93** [comment]
Please ensure you have read the documentation, especially this portion which you were looking for: https://github.com/Cookie-AutoDelete/Cookie-AutoDelete/wiki/Documentation#greylisting

we handle specific cookies a bit differently here.  Only the domain goes under the domain expressions.

First you'll need `*.google.com` under a whitelist, uncheck `Keep All Cookies` and check all the cookies you want to keep but the hsid cookie.
Then you'll need a `*.google.com` under a greylist, uncheck `Keep All Cookies until restart` and only check the hsid cookie and/or any other cookies you want removed at browser startup.

**Reply by snowbound** [reply]
Thanks for the reply. I was mistaken the cookie in question appears to be SID one not the HSID. 

A little history on what I am trying to do. In Chrome the auto-deletion in Google Chrome works properly. I tell it to delete Google.com cookie and upon restart of Chrome, I am logged out of Gmail. However Brave browser which is also a Chromium-based browser has a bug in it that the cookie deletion does not occur. I remain logged in. If I use Cookie Editor extension and delete the SID cookie name under Google.com  and exit Brave I am logged out and have to login as I wanted. During the login process, I do NOT have to enter my 2FA code only my Gmail password.

In Cookie-AutoDelete I found that I could not add Google.com in the whitelist and greylist by entering Google.com in the expression list and clicking on the +Whitelist and then +Greylist. The Greylist entry never showed up. 

I added Google,com into the whitlist and unchecked SID from the Keep All Cookies list. I then had to add gmail.google.com into Greylist and edit the name and removed Gmail portion of the domain.  I then checked only the SID option under Keep All Cookies until Restart, enabled Automatic Cleanup. The result was that I was logged out of Gmail HOWEVER during the login process I also had to enter my 2FA code. 

If I use Cookie Editor extension instead and delete the SID cookie for Google.com close Brave down and start it back up I only have to enter my password not my 2FA code.

**Reply by kennethtran93** [reply]
> In Cookie-AutoDelete I found that I could not add Google.com in the whitelist and greylist by entering Google.com in the expression list and clicking on the +Whitelist and then +Greylist. The Greylist entry never showed up.

Ah yes the feature to have the same domain multiple times is a work in progress.  Until then we only look at the first matched domain expression, whitelist first then greylist.

Normally for the 2FA there should be another cookie that should be set for that if you wanted 2FA not to be asked for that device.

Check the logs to see which cookies were removed by our extension, then you can narrow down which cookie you still need to keep.

**Reply by snowbound** [reply]
I have Enable Greylist Cleanup on Browser restart enabled and Enable Auto Cleanup. I had only the first one enabled and it did not do anything. I have

![image](https://user-images.githubusercontent.com/596147/111228321-00431000-85b2-11eb-8f7b-a87f8357a42f.png)


 
Yet it is touching other domains I do not want it to touch cookies for. I only want the extension to perform the clean up in the above and nothing more.

grammarly.com (gnar_containerId, grauth, csrf-token, experiment_groups, rtMf5RGKqL, yuhoaood48): Clean because of startup cleanup and grammarly.com is not in the White or Grey lists
github.com (_octo, logged_in): Clean because of startup cleanup and github.com is not in the White or Grey lists
google.com (SID, __Secure-3PSID, HSID, SSID, APISID, SAPISID, __Secure-3PAPISID, SIDCC, __Secure-3PSIDCC): Partial clean because of matched *.google.com in the Whitelist
accounts.google.com (LSID): Partial clean because of matched *.google.com in the Whitelist
youtube.com (SID, __Secure-3PSID, HSID, SSID, APISID, SAPISID, __Secure-3PAPISID): Clean because of startup cleanup and youtube.com is not in the White or Grey lists
google.ca (SID, __Secure-3PSID, HSID, SSID, APISID, SAPISID, __Secure-3PAPISID, NID): Clean because of startup cleanup and google.ca is not in the White or Grey lists
raindrop.io (__cfduid): Clean because of startup cleanup and raindrop.io is not in the White or Grey lists
bitwarden.com (__cfduid): Clean because of startup cleanup and bitwarden.com is not in the White or Grey lists
api.raindrop.io (AWSALB, AWSALBCORS): Clean because of startup cleanup and api.raindrop.io is not in the White or Grey lists

**Reply by kennethtran93** [reply]
We do not have specific domain cleanups at this time.  That is a work in progress.

Your partial clean has taken effect, thus the cookies listed are the only ones cleaned in the partial cleanup.

At this time it is only:  
Whitelist:  keep all except unselected (unselected removed on close of last tab with domain).
Greylist:  keep until restart except unselected (unselected removed on close of last tab with domain).

I should add that the _DEFAULT items are templates and will only be used when creating new expressions.  You would need to add other domains to the whitelist to keep data on those sites.

**Reply by snowbound** [reply]
Ok thanks for the help today. Seems that the only way I could only have Gmail being the only domain affected is to whitelist each and every other domain that Brave, in this case, uses in extensions or visits and that is too arduous of a task.

**Reply by kennethtran93** [reply]
You can also try using regular expressions to capture all but one domain, as long as the browser's version is the latest one, or has been released in the last 1-2 years.  Here's one that will capture all but those ending in google.com:  `/(?<!google\.com)$/`.  Try putting that into the whitelist.  Then you can have a separate whitelist for google.com, except check all but the cookie you want to delete.  As long as you still have at least one google.com tab open until browser restart, that cookie you kept unchecked should not be removed until then.

**Reply by snowbound** [reply]
Thanks that seems to have done the trick. Further testing is required on my part on which ones of the cookie names actually have to be deleted on browser restart to produce the desired effect.

**Comment by kennethtran93** [comment, marked as answer]
Repasting as partial solution:

You can also try using regular expressions to capture all but one domain, as long as the browser's version is the latest one, or has been released in the last 1-2 years. Here's one that will capture all but those ending in google.com: /(?<!google\.com)$/. Try putting that into the whitelist. Then you can have a separate whitelist for google.com, except check all but the cookie you want to delete. As long as you still have at least one google.com tab open until browser restart, that cookie you kept unchecked should not be removed until then.

**Reply by snowbound** [reply]
Working fine and as I wanted it to. Appreciate all the help with the issue.
